### PR TITLE
zsh-history-enquirer 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by danielbayley**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Reattempting #122728.

### CODE REVIEW

File: Formula/zsh-history-enquirer.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  homepage "https://zsh-history-enquirer.zthxxx.me"
+  url "https://registry.npmjs.org/zsh-history-enquirer/-/zsh-history-enquirer-1.3.0.tgz"
+  sha256 "d2b673fdb0621bbbbae1529f42ff0c3184fa8e4f4befbc6bd9fa4b2d95ca4b5f"
+  license "MIT"
+
+  depends_on "node"
+  uses_from_macos "expect" => :test
+
+  on_linux do
+    depends_on "expect" => :test
```
Comment (p-linnane) [inline Formula/zsh-history-enquirer.rb:14]:
```suggestion

```

Comment (p-linnane) [reply Formula/zsh-history-enquirer.rb:14]: This is covered by `uses_from_macos` above.

**Review by p-linnane — CHANGES_REQUESTED** [review]

File: Formula/zsh-history-enquirer.rb (hunk 1 above)
Comment (p-linnane) [reply Formula/zsh-history-enquirer.rb:14]: You will need to move this into the `z` directory and also change your commit message to `zsh-history-enquirer 1.3.0 (new formula)`.

**Comment by p-linnane** [comment]
You will need to move this into the `z` directory and squash the commits into one with the message `zsh-history-enquirer 1.3.0 (new formula)`.

**Comment by danielbayley** [comment]
> You will need to move this into the `z` directory and squash the commits into one with the message `zsh-history-enquirer 1.3.0 (new formula)`.

@p-linnane Done.

**Comment by chenrui333** [comment]
I can take a look at the linux build failure tomorrow

**Comment by chenrui333** [comment]
```
/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1
    at makeError$1 (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:1236:11)
    at handlePromise (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2150:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async history (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2315:24)
    at async Promise.all (index 1)
    at async searchHistory (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:6027:30) {
  shortMessage: 'Command failed with exit code 1: /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
  command: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
  escapedCommand: '"/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
runner@fv-az1393-545:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ zsh-history-enquirer
Error: Command failed with exit code 1: /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh
/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1
    at makeError$1 (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:1236:11)
    at handlePromise (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2150:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async history (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2315:24)
    at async Promise.all (index 1)
    at async searchHistory (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:6027:30) {
  shortMessage: 'Command failed with exit code 1: /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
  command: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
  escapedCommand: '"/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

**Comment by danielbayley** [comment]
> ```
> runner@fv-az1393-545:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ zsh-history-enquirer
> Error: Command failed with exit code 1: /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh
> /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1
>     at makeError$1 (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:1236:11)
>     at handlePromise (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2150:26)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async history (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:2315:24)
>     at async Promise.all (index 1)
>     at async searchHistory (/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/index.js:6027:30) {
>   shortMessage: 'Command failed with exit code 1: /home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
>   command: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh',
>   escapedCommand: '"/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh"',
>   exitCode: 1,
>   signal: undefined,
>   signalDescription: undefined,
>   stdout: '',
>   stderr: '/home/linuxbrew/.linuxbrew/Cellar/zsh-history-enquirer/1.3.0/libexec/lib/node_modules/zsh-history-enquirer/dist/history.zsh:fc:13: no such event: 1',
>   failed: true,
>   timedOut: false,
>   isCanceled: false,
>   killed: false
> }
> ```

cc: @zthxxx, as I don’t have a Linux VM set up, or the bandwidth to test…

**Comment by chenrui333** [comment]
you can just use either codespace or  linux action runner + tmate like what I did in here, https://github.com/chenrui333/github-action-test/blob/main/.github/workflows/brew-debug.yml#L49-L67

**Comment by chenrui333** [comment]
@danielbayley can you just file an upstream linux build issue report and we can do `depends_on :macos` for only building bottles for macos for now.

**Comment by danielbayley** [comment]
> @danielbayley can you just file an upstream linux build issue report and we can do `depends_on :macos` for only building bottles for macos for now.

@chenrui333 Sure, filed here: https://github.com/zthxxx/zsh-history-enquirer/issues/73.

**Comment by zthxxx** [comment]
> `depends_on :macos`

It can run exactly under Linux Zsh; I also use it on all of my servers.

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by cho-m** [comment]
Tried some things out but still failing in CI.

In local Linux container, adjusting match seemed to help as arrow symbol was rendering differently.

Also, when manually running with `^R`, command would fail without some basic ZSH history configs. However, I don't think this would impact existing test, but added fake `.zshrc` to try out.

Remaining things would be CI specific settings (perhaps `TERM` settings and so on).

**Comment by danielbayley** [comment]
> It can run exactly under Linux Zsh; I also use it on all of my servers.
> 
> I noticed that the CI has failed due to the _unit-test_. Is that the problem with the test scripts?

@zthxxx Sounds like it must be. Maybe you want to look into getting it working with `brew test` on your local Linux setup?

**Comment by zthxxx** [comment]
@danielbayley  I'm really sorry about this. A few months ago, I have tried to solve testing on this, but I failed due to I'm not familiar with Ruby or how Homebrew tests running. 😢

**Pull request closed without merging**